### ISSUE TITLE
Double Alerts

### ISSUE THREAD

**Issue by ge8020** [briis/weatherbit#55]
Hello!
There has been an issue on this topic before, but it´s closed...
I also have double alerts on almost every refresh (here, alert 1 and 3)->
--------------------------------------------------------------------------------------------------
Attribute
Alerts
- Alert No 1: '-------------------'
title: Wind
severity: Advisory
effective: '2022-01-13T21:00:00'
onset: '2022-01-13T21:00:00'
ends: '2022-01-14T17:00:00'
expires: '2022-01-14T17:00:00'
uri: https://meteoalarm.org?geocode=EMMA_ID:AT009
city_name: Rothleiten
regions:
- Steiermark,AT
description_english: >-
Gusts of 70 kph are possible. BE AWARE of debris being blown around.
Localised outdoor activity disruption is possible due to debris.
description_local: >-
Starker Wind mit Spitzen bis 70 km/h ist zu erwarten. VORSICHT, durch den
starken Wind können lose Gegenstände herumfliegen. Tätigkeiten im Freien
können beeinträchtigt sein.
- Alert No 2: '-------------------'
title: Extreme Low Temperature
severity: Advisory
effective: '2022-01-13T23:00:00'
onset: '2022-01-13T23:00:00'
ends: '2022-01-14T22:59:59'
expires: '2022-01-14T22:59:59'
uri: https://meteoalarm.org?geocode=EMMA_ID:AT009
city_name: Rothleiten
regions:
- Steiermark,AT
description_english: >-
Increased cold stress must be expected. BE AWARE that low temperatures are
expected. Some health risks amongst vulnerable people e.g. the elderly, very
young and homeless are possible.
description_local: >-
Es ist mit erhöhter Kältebelastung zu rechnen. VORSICHT, es wird sehr kalt.
Auswirkungen auf die Gesundheit sind nicht ausgeschlossen, z.B. bei alten
Menschen, Kindern und Obdachlosen.
- Alert No 3: '-------------------'
title: Wind
severity: Advisory
effective: '2022-01-13T21:00:00'
onset: '2022-01-13T21:00:00'
ends: '2022-01-14T17:00:00'
expires: '2022-01-14T17:00:00'
uri: https://meteoalarm.org?geocode=EMMA_ID:AT009
city_name: Rothleiten
regions:
- Steiermark,AT
description_english: >-
Gusts of 70 kph are possible. BE AWARE of debris being blown around.
Localised outdoor activity disruption is possible due to debris.
description_local: >-
Starker Wind mit Spitzen bis 70 km/h ist zu erwarten. VORSICHT, durch den
starken Wind können lose Gegenstände herumfliegen. Tätigkeiten im Freien
können beeinträchtigt sein.
--------------------------------------------------------------------------------------------
when i take a look at meteoalarm (which seems to be the source for the alarms at my adress), there are only two alarms in this case...
is there a way to check, if weatherbit.io is making troubles when passing on this alerts?

**Comment by briis** [comment]
It is the data I get from Weatherbit.io that is shown here, and it seems they have some kind of error, sending alerts more than 1 time. I will need to see if there is any way I can do a comparison between the alerts before adding them to the array.
The issue here is that alerts disappear within a timeframe, so I have to be lucky to get the raw data while there is an error, to test this. (And I am from Denmark, and never get any alerts 🤔

**Comment by ge8020** [comment]
Here in Austria i have alerts all the time, but this is nothing unusual, especially this time of year :-) Tell me, when i can help in any way. Thx!

**Comment by briis** [comment]
Hi @ge8020 
Can you let me know next time you have the double alerts, so that I can capture the data.

**Comment by ge8020** [comment]
hello @briis !
Okay, i will tell you next time a double alert happens. At this time i have no alerts. Should i reply here in this thread?

**Comment by briis** [comment]
Yes just reply here, I get a notification when you do.

**Comment by ge8020** [comment]
Hello, just got a double alert again (1+3): -->

Attribute
Alerts
- Alert No 1: '-------------------'
title: Snow/Ice
severity: Watch
effective: '2022-01-31T11:00:00'
onset: '2022-01-31T11:00:00'
ends: '2022-02-02T23:00:00'
expires: '2022-02-02T23:00:00'
uri: https://meteoalarm.org?geocode=EMMA_ID:AT009
city_name: Rothleiten
regions:
- Steiermark,AT
description_english: >-
Fresh snow up to 80 cm is possible. BE PREPARED for widespread snow and/or
ice leading to disruption to road, rail and air transport and outdoor
activities.
description_local: >-
Neuschnee bis 80 cm ist zu erwarten. ACHTUNG, rechnen Sie mit Problemen
durch Schnee und/oder Eis auf Straßen und Wegen. Auch beim Bahn- und
Flugverkehr sowie bei Tätigkeiten im Freien sind Beeinträchtigungen möglich.
- Alert No 2: '-------------------'
title: Wind
severity: Advisory
effective: '2022-01-31T11:00:00'
onset: '2022-01-31T11:00:00'
ends: '2022-01-31T21:00:00'
expires: '2022-01-31T21:00:00'
uri: https://meteoalarm.org?geocode=EMMA_ID:AT009
city_name: Rothleiten
regions:
- Steiermark,AT
description_english: >-
Gusts of 70 kph are possible. BE AWARE of debris being blown around.
Localised outdoor activity disruption is possible due to debris.
description_local: >-
Starker Wind mit Spitzen bis 70 km/h ist zu erwarten. VORSICHT, durch den
starken Wind können lose Gegenstände herumfliegen. Tätigkeiten im Freien
können beeinträchtigt sein.
- Alert No 3: '-------------------'
title: Snow/Ice
severity: Watch
effective: '2022-01-31T11:00:00'
onset: '2022-01-31T11:00:00'
ends: '2022-02-02T23:00:00'
expires: '2022-02-02T23:00:00'
uri: https://meteoalarm.org?geocode=EMMA_ID:AT009
city_name: Rothleiten
regions:
- Steiermark,AT
description_english: >-
Fresh snow up to 80 cm is possible. BE PREPARED for widespread snow and/or
ice leading to disruption to road, rail and air transport and outdoor
activities.
description_local: >-
Neuschnee bis 80 cm ist zu erwarten. ACHTUNG, rechnen Sie mit Problemen
durch Schnee und/oder Eis auf Straßen und Wegen. Auch beim Bahn- und
Flugverkehr sowie bei Tätigkeiten im Freien sind Beeinträchtigungen möglich.

**Comment by briis** [comment]
Perfect. I have just captured the raw data, and will look at it.

**Comment by briis** [comment]
Weatherbit delivers 3 alerts, so technically the error lies with them. I will analyse the data a bit to see if I can filter out double alerts without to much hazzle. Here is the raw data delivered:

````
[
    {'regions': ['Steiermark,AT'], 
        'ends_utc': '2022-02-02T23:00:00', 
        'effective_local': '2022-01-31T12:00:00', 
        'onset_utc': '2022-01-31T11:00:00', 
        'expires_local': '2022-02-03T00:00:00', 
        'expires_utc': '2022-02-02T23:00:00', 
        'ends_local': '2022-02-03T00:00:00', 
        'uri': 'https://meteoalarm.org?geocode=EMMA_ID:AT009', 
        'onset_local': '2022-01-31T12:00:00', 
        'effective_utc': '2022-01-31T11:00:00', 
        'severity': 'Watch', 
        'title': 'Snow/Ice', 
        'description': 'German(de-DE): Neuschnee von 30 bis 80 cm ist zu erwarten. ACHTUNG, rechnen Sie mit Problemen durch Schnee und/oder Eis auf Straßen und Wegen. Auch beim Bahn- und Flugverkehr sowie bei Tätigkeiten im Freien sind Beeinträchtigungen möglich.\nEnglish(en-GB): Fresh snow between 30 and 80 cm is possible. BE PREPARED for widespread snow and/or ice leading to disruption to road, rail and air transport and outdoor activities.'}, 
    {'regions': ['Steiermark,AT'], 
        'ends_utc': '2022-01-31T21:00:00', 
        'effective_local': '2022-01-31T12:00:00', 
        'onset_utc': '2022-01-31T11:00:00', 
        'expires_local': '2022-01-31T22:00:00', 
        'expires_utc': '2022-01-31T21:00:00', 
        'ends_local': '2022-01-31T22:00:00', 
        'uri': 'https://meteoalarm.org?geocode=EMMA_ID:AT009', 
        'onset_local': '2022-01-31T12:00:00', 
        'effective_utc': '2022-01-31T11:00:00', 
        'severity': 'Advisory', 
        'title': 'Wind', 
        'description': 'German(de-DE): Starker Wind mit Spitzen bis 70 km/h ist zu erwarten. VORSICHT, durch den starken Wind können lose Gegenstände herumfliegen. Tätigkeiten im Freien können beeinträchtigt sein.\nEnglish(en-GB): Gusts of 70 kph are possible. BE AWARE of debris being blown around. Localised outdoor activity disruption is possible due to debris.'}, 
    {'regions': ['Steiermark,AT'], 
        'ends_utc': '2022-02-02T23:00:00', 
        'effective_local': '2022-01-31T12:00:00', 
        'onset_utc': '2022-01-31T11:00:00', 
        'expires_local': '2022-02-03T00:00:00', 
        'expires_utc': '2022-02-02T23:00:00', 
        'ends_local': '2022-02-03T00:00:00', 
        'uri': 'https://meteoalarm.org?geocode=EMMA_ID:AT009', 
        'onset_local': '2022-01-31T12:00:00', 
        'effective_utc': '2022-01-31T11:00:00', 
        'severity': 'Watch', 
        'title': 'Snow/Ice', 
        'description': 'German(de-DE): Neuschnee von 30 bis 80 cm ist zu erwarten. ACHTUNG, rechnen Sie mit Problemen durch Schnee und/oder Eis auf Straßen und Wegen. Auch beim Bahn- und Flugverkehr sowie bei Tätigkeiten im Freien sind Beeinträchtigungen möglich.\nEnglish(en-GB): Fresh snow between 30 and 80 cm is possible. BE PREPARED for widespread snow and/or ice leading to disruption to road, rail and air transport and outdoor activities.'}
]
````

**Comment by briis** [comment]
Could you try and download the file `manifest.json` from this Github and replace the file you have in `custom_components/weatherbit` on your HA installation. Then restart HA and see if it works?

I am now comparing severity, title and end end_time and if they are already present, I am not adding the alert. Seems to work for me.

**Comment by ge8020** [comment]
Thanks for your effort!
Just copied the new manifest.json, but at this time i have 3 different alerts at the same time, maybe we get some snow again ;-)
I made a notification automation everytime the number of alerts changes and will have a look at it.
Thank you!

**Comment by briis** [comment]
I saw that too. But if you look carefully, two of the alerts a regarding Wind and the Text is different in them. So they are different. But most likely one of them should replace the other one, but has not been pulled yet.
I don't think there is much more I can do on this than I do now.

**Comment by briis** [comment]
I will close this issue now. The above fix will be in the 1.0.9 release. If it shows up again, please re-open
